### ISSUE TITLE
Snowflake: Create Table does not support In Line Foreign Key Constraints

### ISSUE THREAD

**Issue by WobblyRobbly** [sqlfluff/sqlfluff#6485]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This report follows on from #6473 

Whilst attempting to `lint` the following DDL

```sql
CREATE OR REPLACE TABLE SOME_TABLE1 (
    ID INTEGER NOT NULL CONSTRAINT MY_FK
    FOREIGN KEY REFERENCES ADDRESSES (ADDRESS)
    ON DELETE CASCADE
);

CREATE OR REPLACE TABLE SOME_TABLE2 (
    ID INTEGER NOT NULL,
    CONSTRAINT MY_FK FOREIGN KEY (ID) REFERENCES ADDRESSES (ADDRESS)
    ON DELETE CASCADE
);
```

The first statement with inline foreign key fails to lint.

```cmd
❯ sqlfluff lint --dialect snowflake create_table.sql
❯ sqlfluff lint --dialect snowflake create_table.sql
== [create_table.sql] FAIL
L:   1 | P:  37 |  PRS | Line 1, Position 37: Found unparsable section: '(\n
                       | ID INTEGER NOT NULL CONSTRAINT MY_...'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

The second statement lints successfully

### Expected Behaviour

The lint command should run successfully and not report unparseable

### Observed Behaviour

The following error was reported

```cmd
❯ sqlfluff lint --dialect snowflake create_table.sql
== [create_table.sql] FAIL
L:   6 | P:  36 |  PRS | Line 6, Position 36: Found unparsable section: '(\n
                       | ID INTEGER NOT NULL CONSTRAINT MY_...'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Add the DDL above into `create_table.sql` and execute

```cmd
sqlfluff lint --dialect snowflake create_table.sql
```

### Dialect

snowflake

### Version

3.2.5

### Configuration

files do not exist

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
